src/main/scala/chiseltest/stage/phases/AddDefaults.scala

```diff
@@ -0,0 +1,48 @@
+package chiseltest.stage.phases
+
+import chiseltest.stage._
+import firrtl.AnnotationSeq
+import firrtl.options.{Dependency, Phase, TargetDirAnnotation}
+
+/** add default [[TargetDirAnnotation]] and [[BackendAnnotation]]. */
+class AddDefaults extends Phase {
+  override def prerequisites: Seq[Dependency[Phase]] = Seq(Dependency[Checks])
+
+  override def optionalPrerequisites = Seq.empty
+
+  override def optionalPrerequisiteOf = Seq.empty
+
+  override def invalidates(a: Phase) = false
+
+  override def transform(annotations: AnnotationSeq): AnnotationSeq = {
+
+    /** Get TargetDirAnnotation and convert it to absolute path to make simulators happier. */
+    val targetDirAnnotation = TargetDirAnnotation(
+      new java.io.File(annotations.collectFirst {
+        /* https://github.com/freechipsproject/firrtl/issues/1929 */
+        case TargetDirAnnotation(".") =>
+          annotations.collectFirst {
+            case TestNameAnnotation(name) => TargetDirAnnotation(s"test_run_dir" + java.io.File.separator + name)
+          }.get
+        case t: TargetDirAnnotation => t
+      }.getOrElse(TargetDirAnnotation(s"test_run_dir" + java.io.File.separator + name)).directory).getAbsolutePath
+    )
+
+    val backendAnnotation = annotations.collect {
+      case t: TreadleBackendAnnotation   => t
+      case t: VerilatorBackendAnnotation => t
+      case t: VcsBackendAnnotation       => t
+    }
```

Comment:
Can you instead collect the common supertype (`BackendAnnotation` or something?)?